pkg/vmprovider/providers/vsphere2/network/network.go

```diff
@@ -259,30 +260,42 @@ func createNetOPNetworkInterface(
 		return nil, fmt.Errorf("network kind %q is not supported for VDS", kind)
 	}
 
+	var err error
 	// If empty, NetOP will try to select a namespace default.
 	networkName := interfaceSpec.Network.Name
+	netIf := &netopv1alpha1.NetworkInterface{}
+	netIfKey := types.NamespacedName{
+		Namespace: vmCtx.VM.Namespace,
+		Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, true),
+	}
 
-	netIf := &netopv1alpha1.NetworkInterface{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, false),
-			Namespace: vmCtx.VM.Namespace,
-		},
+	// check if a networkIf object exists with the older (v1a1) naming convention
+	if err = client.Get(vmCtx, netIfKey, netIf); ctrlruntime.IgnoreNotFound(err) != nil {
+		return nil, err
 	}
 
-	_, err := controllerutil.CreateOrUpdate(vmCtx, client, netIf, func() error {
```

Comment:
Still always call the `CreateOrUpdate`. We want to make sure that this VM is the controlling owner of the NIC CR before using it.